src/librustc/ty/context.rs

```diff
@@ -1179,6 +1179,10 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
         self.all_crate_nums(LOCAL_CRATE)
     }
 
+    pub fn cstore(&self) -> &CrateStore {
```

Comment:
I think @alexcrichton removed this for a reason.

Reply:
I'm sure. This was a quick fix after rebasing against master to get the code _as I had written it_ working. I fully intend to clean this up before attempting to merge this PR.